### DISCUSSION TITLE
Access existing OpenAI Assistant through API via Assistant ID

### DISCUSSION THREAD

**Discussion by solaius** [mckaywrigley/chatbot-ui#1256] [Ideas]
Hello All!

Is there a way of connecting to an already created OpenAI GPT assistants via their ID instead of directly to the full model, like you would with AutoGen and LangChain? I did not see anything in the documentation around this.  

Example: 
researcher = GPTAssistantAgent(
    name = "researcher",
    llm_config = {
        "config_list": config_list,
        "assistant_id": "asst_XXXXXXXXXXXXXXXXXXXXXXX"
    }
)

**Comment by jsweetzer-ea** [comment]
@mckaywrigley You mentioned this was coming soon - do you have any updates on this? Thank you!

**Comment by MeanTemperature** [comment]
This is why I'm in this repo.  Anyone got it to work?

**Comment by princemuichkine** [comment]
I believe this is now irrelevant considering the amount of tools online to enable this use case.

On 13 Mar 2025, at 12:03 AM, JustinHeath ***@***.***> wrote:



This is why I'm in this repo. Anyone got it to work?

**Reply by MeanTemperature** [reply]
Can you suggest some?  My use case is that I have an OpenAI assistant that is multi-modal and I need a frontend that other people can access it through, but on their own accounts. I don't want my API key being used in their browser, so need a backend.  This looked like a great possibility.

**Comment by kinthaiofficial** [comment]
Accessing existing OpenAI Assistants via ID is useful for continuity scenarios — but it surfaces a deeper question about where conversation state lives.

The Assistants API stores thread state server-side, which is convenient but creates a few problems:

**Portability** — the thread is locked to OpenAI. If you switch providers (Anthropic, local model), the thread state doesn't transfer. For agent systems that need model routing across providers, this is a hard dependency.

**Cost attribution** — Assistants API doesn't expose per-message cost breakdown. The thread is a black box from a billing perspective, which makes it hard to know which parts of a long conversation are expensive.

**State ownership** — the conversation history lives in OpenAI's systems, not yours. For anything privacy-sensitive or compliance-relevant, you want conversation state stored in infrastructure you control.

The pattern we use in KinthAI: store conversation/session state client-side (our DB), pass context to the model on each call explicitly. This is more expensive (more tokens per call) but gives full portability, full cost visibility, and full state ownership.

For accessing existing threads specifically: if the thread has long history, you may want to summarize + compress it before passing to a new context, rather than replaying the full history. There's a write-up on context compaction strategies here: https://blog.kinthai.ai/why-character-ai-forgets-you-persistent-memory-architecture — covers the tradeoffs between full history, structured summaries, and digest.

Is the cross-session continuity need mainly for personal use cases, or for multi-user shared assistants?
